examples/bank/bank.go

```diff
@@ -117,43 +122,36 @@ func (bank *Bank) continuousMoneyTransfer(cash int64) {
 			if err := runner.Run(batchRead); err != nil {
 				return err
 			}
-			if batchRead.Results[0].Err != nil {
-				return batchRead.Results[0].Err
-			}
+			var err error
 			// Read from value.
 			fromAccount := &Account{}
-			err := fromAccount.decode(batchRead.Results[0].Rows[0].ValueBytes())
-			if err != nil {
-				return err
-			}
-			// Ensure there is enough cash.
-			if fromAccount.Balance < exchangeAmount {
-				return nil
-			}
+			fromAccount.decode(batchRead.Results[0].Rows[0].ValueBytes(), &err)
 			// Read to value.
 			toAccount := &Account{}
-			errRead := toAccount.decode(batchRead.Results[0].Rows[1].ValueBytes())
-			if errRead != nil {
-				return errRead
-			}
+			toAccount.decode(batchRead.Results[0].Rows[1].ValueBytes(), &err)
 			// Update both accounts.
-			batchWrite := &client.Batch{}
 			fromAccount.Balance -= exchangeAmount
 			toAccount.Balance += exchangeAmount
-			if fromValue, err := fromAccount.encode(); err != nil {
-				return err
-			} else if toValue, err := toAccount.encode(); err != nil {
+			fromValue := fromAccount.encode(&err)
```

Comment:
I actually did put it into the Account struct originally and that's a bit
of an improvement but not quite. Let me put that back so you can see that.
This may still not be worth checking in

On Mon, May 18, 2015 at 3:35 PM MBerhault <notifications@github.com> wrote:

> In examples/bank/bank.go
> <https://github.com/cockroachdb/cockroach/pull/1048#discussion_r30539878>:
>
> >  			fromAccount.Balance -= exchangeAmount
> >  			toAccount.Balance += exchangeAmount
> > -			if fromValue, err := fromAccount.encode(); err != nil {
> > -				return err
> > -			} else if toValue, err := toAccount.encode(); err != nil {
> > +			fromValue := fromAccount.encode(&err)
>
> yes, but you're still using data that was generated by failing code. it
> just so happens that 1) your struct is initialized so you won't have nils,
> and 2) encode is happy with an empty struct. This is rarely the case.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/cockroachdb/cockroach/pull/1048/files#r30539878>.
>